mlir/lib/Dialect/Vector/Transforms/LowerVectorTransfer.cpp

```diff
@@ -269,7 +280,7 @@ struct TransferWriteNonPermutationLowering
                                     missingInnerDim.size());
     // Mask: add unit dims at the end of the shape.
     Value newMask;
-    if (op.getMask())
+    if (op.getMask() && !op.getVectorType().isScalable())
```

Comment:
Why can the mask be omitted for scalable vectors? I see the input test case is also masked, but the new `vector.transfer_write` is unmasked.

Reply:
We can erase the unsupported transpose. Similar to how we do it TransferWritePermutationLowering?

Reply:
In `TransferWritePermutationLowering` the rewrite fails (i.e. it does nothing), erasing the transpose is just doing a cleanup. Here the rewrite succeeds but the mask is omitted, which changes the semantics of the operation, which could lead to incorrect results.